packages/block-editor/src/components/block-heading-level-dropdown/index.js

```diff
@@ -40,12 +40,18 @@ export default function HeadingLevelDropdown( {
 	value,
 	onChange,
 } ) {
+	const validOptions = options
+		.filter(
+			( option ) => option === 0 || HEADING_LEVELS.includes( option )
+		)
+		.sort( ( a, b ) => a - b ); // Sorts numerically in ascending order;
```

Comment:
Note: While usually sorting callback can be omitted in similar cases, this is safer when dealing with numeric values.